File: app/src/main/java/shinil/direct_share/MainActivity.java
package shinil.direct_share;

import android.os.Bundle;
import android.widget.TextView;

import shinil.direct.share.DirectNetShare;
import shinil.direct.share.util.Constants;

public class MainActivity extends BaseActivity
        implements DirectNetShare.GroupCreatedListener {

    private DirectNetShare share;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void startShare() {
        share = new DirectNetShare(MainActivity.this, this);
        share.start();
    }

    @Override
    public void onGroupCreated(String ssid, String password) {
        ((TextView) findViewById(R.id.text_view_info)).setText("SSID : "+ssid+"\nPassword : "+password);
        ((TextView) findViewById(R.id.text_view_hint)).setText("After connecting, set the proxy settings to" +
                "\nhost : "+ Constants.DEFAULT_GROUP_OWNER_IP+"\nport : "+Constants.PROXY_PORT+"\non the other device.");
    }

    @Override
    protected void onPause() {
        super.onPause();
        share.stop();
    }
}

File: app/src/main/java/shinil/direct_share/MainActivity.kt
package shinil.direct_share

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.SwitchCompat
import android.util.Log
import android.widget.CompoundButton
import android.widget.TextView
import kotlinx.coroutines.experimental.launch
import shinil.direct.share.DirectNetShare

class MainActivity : AppCompatActivity() {
    private lateinit var infoTv: TextView
    private val rootManager = RootManager()
    private val groupCreatedListener = DirectNetShare.GroupCreatedListener { ssid, password -> infoTv.text = String.format("SSID : %s\nPassword : %s", ssid, password) }
    private lateinit var share: DirectNetShare
    private val onCheckedChangeListener = CompoundButton.OnCheckedChangeListener { _, isChecked ->
        if (isChecked) {
            checkWifiAndStart()
        } else {
            stopShare()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        infoTv = findViewById(R.id.info)
        (findViewById<SwitchCompat>(R.id.wifi_switch)).setOnCheckedChangeListener(onCheckedChangeListener)
    }

    override fun onDestroy() {
        super.onDestroy()
        stopShare()
    }

    private fun checkWifiAndStart() {
        if (!Utils.isWifiEnabled(applicationContext)) {
            registerReceiver(object : BroadcastReceiver() {
                override fun onReceive(context: Context, intent: Intent) {
                    if (intent.action != null && intent.action == ConnectivityManager.CONNECTIVITY_ACTION) {
                        val noConnectivity = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false)
                        if (!noConnectivity) {
                            startShare()
                            unregisterReceiver(this)
                        }
                    }
                }
            }, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
            Utils.enableWifi(applicationContext)
        } else {
            startShare()
        }
    }

    private fun startShare() {
        share = DirectNetShare(this@MainActivity, groupCreatedListener)
        share.start()
        launch {
            val success = rootManager.dhcpSetup()
            Log.d("MainActivity", "DHCP setup successful? $success")
        }
    }

    private fun stopShare() {
        share.stop()
        infoTv.text = ""
    }
}
